src/api.rs

```diff
@@ -86,49 +93,49 @@ pub fn get_peer_count() -> Result<u32, String> {
         .map_err(|e| e.to_string())
 }
 
-pub fn scan_next_n_blocks(n: u32) -> Result<(), String> {
-    let sp_client = get_sp_client();
-
-    let sp_receiver = &sp_client.sp_receiver;
-    let scan_sk = sp_client.scan_sk;
-
+pub fn scan_next_n_blocks(path: String, label: String, n: u32) -> Result<(), String> {
     let electrum_client = create_electrum_client()
         .map_err(|e| e.to_string())?;
 
-    let scan_key_scalar: Scalar = scan_sk.into();
-
-    nakamotoclient::scan_blocks(n, sp_receiver, electrum_client, scan_key_scalar)
-        .map_err(|e| e.to_string())
+    match SpClient::try_init_from_disk(label, path) {
+        Ok(sp_client) => nakamotoclient::scan_blocks(n, sp_client, electrum_client).map_err(|e| e.to_string())?,
+        Err(_) => return Err("Wallet not found".to_owned())
+    }
+    Ok(())
 }
 
-pub fn scan_to_tip() -> Result<(), String> {
+pub fn scan_to_tip(path: String, label: String) -> Result<(), String> {
     // 0 means scan to tip
-    scan_next_n_blocks(0)
+    scan_next_n_blocks(path, label, 0)
 }
 
-pub fn get_wallet_info() -> Result<WalletStatus, String> {
-    let scanheight = db::get_scan_height()
-        .map_err(|e| e.to_string())?;
+pub fn get_wallet_info(path: String, label: String) -> Result<WalletStatus, String> {
+    let mut sp_client: SpClient;
+    match SpClient::try_init_from_disk(label, path) {
+        Ok(s) => sp_client = s,
+        Err(_) => return Err("Wallet not found".to_owned())
+    }
+
+    let scan_height = sp_client.last_scan;
+    let birthday = sp_client.birthday;
     let tip_height = nakamotoclient::get_tip()
         .map_err(|e| e.to_string())?;
-    let amount = get_wallet_balance()?;
+    let amount = sp_client.get_total_amt();
 
     Ok(WalletStatus {
         amount,
-        scan_height: scanheight,
-        block_tip: tip_height,
+        birthday,
+        scan_height,
+        tip_height,
     })
 }
 
-pub fn get_birthday() -> u32 {
-    spclient::get_birthday()
-}
-
-pub fn get_wallet_balance() -> Result<u32, String> {
-    db::get_sum_owned()
-        .map_err(|e| e.to_string())
-}
+pub fn get_receiving_address(path: String, label: String) -> Result<String, String> {
```

Comment:
This is a general comment for every function that has `path` and `label` as arguments: I think adding `label` as an argument makes sense, this would allow us to have multiple wallets in memory later on. But I think it's better if we let the user decide the `path` only once, for example when creating the `sp_client` object

Reply:
I don't think that's a good idea on mobile at least, afaiu there is Internal Storage which is standard and safe and that's system dependent. Or we take a different approach and say that wallet is saved on External Storage and then yes it makes sense to let the user decide, the downside being that we save private materials and External Storage can be read by other applications. 

If on target we encrypt the private data it's possible though. 

One upside of using External Storage is that it becomes easier for user to move around the wallet file for backuping and changing device (ok that's maybe not an upside from a security pov). 

Maybe the best is to use Internal Storage (safer + not bothering user with choosing a path for his files), but then allow user to export an encrypted backup to External Storage (encryption on Internal would only be for private keys and use a simple input like a PIN from user, while exporting to External will encrypt the whole file with a stronger password)

Reply:
Yes sorry, I wasn't clear in my original post. I would prefer Internal Storage as well, but I mean that we should change it so that in e.g. the `setup` function, we set the `path`, so we don't have to pass it for every call that requires the `sp_client`

Reply:
Ok it makes sense, I'll think about that